src/lib.rs

```diff
@@ -0,0 +1,420 @@
+pub trait Trait: system::Trait {
+
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    //Security/configuration constraints
+    type PostTitleMaxLength: Get<MaxLength>;
+
+    type PostBodyMaxLength: Get<MaxLength>;
+
+    type ReplyMaxLength: Get<MaxLength>;
+
+    type PostsMaxNumber: Get<MaxNumber>;
+
+    type RepliesMaxNumber: Get<MaxNumber>;
+
+    type DirectRepliesMaxNumber: Get<MaxNumber>;
+
+    type ConsecutiveRepliesMaxNumber: Get<MaxConsecutiveRepliesNumber>;
+
+    /// Type of identifier for blogs.
+    type BlogId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    /// Type of identifier for blog posts.
+    type PostId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    /// Type of identifier for replies.
+    type ReplyId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Post {
+    title: Vec<u8>,
+    body: Vec<u8>,
+}
+
+impl Post {
+    fn new(title: Vec<u8>, body: Vec<u8>) -> Self {
+        Self { title, body }
+    }
+}
+
+// Blog`s pallet storage items.
+decl_storage! {
+    trait Store for Module<T: Trait> as BlogModule {
+
+        // Blog Ids set, associated with owner
+        BlogIds get(fn blog_ids_by_owner): map T::AccountId => Option<BTreeSet<T::BlogId>>;
+
+        // Post Ids set, associated with blog
+        BlogPostIds get(fn post_ids_by_blog_id): map T::BlogId => Option<BTreeSet<T::PostId>>;
+
+        // Root replies
+        BlogPostParentReplyIds get (fn post_root_reply_ids): map (T::BlogId, T::PostId) => Option<BTreeSet<T::ReplyId>>;
+
+        // Direct replies to reply
+        BlogPostChildReplyIds get (fn post_child_reply_ids): map (T::BlogId, T::PostId, T::ReplyId) => Option<BTreeSet<T::ReplyId>>;
+
+        BlogPost get(fn post_by_id): map (T::BlogId, T::PostId) => Option<Post>;
+
+        Reply get (fn reply_by_id): map (T::BlogId, T::PostId, T::ReplyId) => Option<Vec<u8>>;
+
+        BlogLockedStatus get(fn blog_locked): map T::BlogId => bool;
+
+        PostLockedStatus get(fn post_locked): map (T::BlogId, T::PostId) => bool;
+
+        //Reply Ids set, associated with owner
+        ReplyIds get (fn reply_ids_by_owner): map T::AccountId => Option<BTreeSet<(T::BlogId, T::PostId, T::ReplyId)>>;
+
+        BlogsCount get(fn blogs_count): T::BlogId;
+
+        // Overall post count by blog
+        PostsCount get(fn posts_count): map T::BlogId => T::PostId;
+
+        // Overall replies count by blog
+        RepliesCount get(fn replies_count): map (T::BlogId, T::PostId) => T::ReplyId;
```

Comment:
Should be value in `struct Reply`.

Reply:
Agree

Reply:
Done here https://github.com/iorveth/substrate-blog-module/commit/a682cc3630bf28aa110ac0c0186936885c3fc2e3#diff-b4aea3e418ccdb71239b96952d9cddb6R190
Also switched to double_map `ReplyById` storage representation
https://github.com/Joystream/substrate-blog-module/pull/1/commits/9a9b38811282a69c9d37071aef4845325d7fe519#diff-b4aea3e418ccdb71239b96952d9cddb6R222